### PR TITLE
gpu: Cleanup error message constants

### PR DESCRIPTION
**Pull request by spencer-lunarg**
This cleans up the mess of how we report back which error GPU-AV found

comments inlined

### CODE REVIEW

File: BUILD.gn [hunk 1]
```diff
@@ -155,6 +155,7 @@ vvl_sources = [
   "layers/external/vma/vk_mem_alloc.h",
   "layers/external/vma/vma.cpp",
   "layers/external/vma/vma.h",
+  "layers/gpu_shaders/gpu_error_constants.h",
```
Comment (spencer-lunarg) [inline BUILD.gn:158]: didn't want to start making too many extra files, but it seems logical to split the error constant with the rest of the "misc" constants

File: layers/gpu_shaders/gpu_error_constants.h [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+// Size of Common and Stage-specific Members
+const int kInstStageOutCnt = kInstCommonOutCnt + 3;
+
+// This identifies the validation error
+// We use groups to more easily mangage the many int values not conflicting
+const int kErrorGroup = kInstStageOutCnt;
+const int kErrorSubCode = kErrorGroup + 1;
+
+// Maximum Output Record Member Count
+const int kInstMaxOutCnt = kErrorSubCode + 6;
```
Comment (spencer-lunarg) [inline layers/gpu_shaders/gpu_error_constants.h:59]: everything from here and above groups the "where in the buffer" offsets, helpful having it next to each other

File: layers/gpu_validation/gpu_error_message.cpp [hunk 3]
```diff
@@ -554,8 +568,23 @@ bool gpuav::Validator::AnalyzeAndGenerateMessages(VkCommandBuffer cmd_buffer, Vk
 
     std::string error_msg;
     std::string vuid_msg;
-    const bool error_found =
-        GenerateValidationMessage(debug_record, cmd_resources, descriptor_sets, error_msg, vuid_msg, oob_access);
```
Comment (spencer-lunarg) [inline layers/gpu_validation/gpu_error_message.cpp:558]: after turning this function into a single switch case that called other functions, it made no sense to have it not just be directly called here (and only pass the extra params to each `LogMessageInst*` as needed)

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

**Pull request merged**